### PR TITLE
simulate option

### PR DESCRIPTION
**Pull request by graykode**

### CODE REVIEW

File: utils/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export async function writeContract(
+  publicClient: PublicClient,
+  walletClient: GetWalletClientResult,
+  args: WriteContractParameters | SimulateContractParameters,
+): Promise<Hash | undefined> {
+  if (!walletClient) {
+    return
+  }
+  const useSimulator =
```
Comment (onemonster) [inline utils/wallet.ts:17]: `useSimulate` sounds more natural than `useSimulator`.
`useSimulate` should default to true as that's what the production build will use.
The boolean can be simplified like so. 
```
const useSimulate = !(process.env.NEXT_PUBLIC_USE_SIMULATE === 'false')
```

File: utils/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    process.env.NEXT_PUBLIC_USE_SIMULATOR !== undefined
+      ? process.env.NEXT_PUBLIC_USE_SIMULATOR === 'true'
+      : false
+  if (useSimulator) {
+    const { request } = await publicClient.simulateContract(
+      args as SimulateContractParameters,
+    )
+    const hash = await walletClient.writeContract(request)
```
Comment (onemonster) [inline utils/wallet.ts:25]: `return walletClient.writeContract(request)`

File: utils/wallet.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return hash
+  } else {
+    const hash = await walletClient.writeContract(
```
Comment (onemonster) [inline utils/wallet.ts:28]: return walletClient.writeContract(args as WriteContractParameters)

File: utils/wallet.ts (hunk 1 above)
Comment (graykode) [reply utils/wallet.ts]: https://github.com/clober-dex/coupon.finance/pull/110/commits/66ac336e9ff5f9b7d6d72027a802593c00a8cf70

File: utils/wallet.ts (hunk 2 above)
Comment (graykode) [reply utils/wallet.ts]: https://github.com/clober-dex/coupon.finance/pull/110/commits/6beb41073ab46057881d98643300d2bd57601ca4

Comment (graykode) [reply utils/wallet.ts]: Done!

**Review by onemonster — APPROVED** [review]

**Pull request merged**